portfolio/src/main/webapp/script.js

```diff
@@ -56,3 +56,25 @@ function getRandomName() {
     document.getElementById('name-container').innerText = name;
   });
 }
+
+/**
+ * Fetches names from DataServlet and adds them to the DOM as a list.
+ */
+function getNames() {
+  fetch('/data').then(response => response.json()).then((names) => {
```

Comment:
I think the servlet you configured replies to any get request with the names, not just to 'GET /data' right?

e.g. this could be '/data' or '/whatever' and it would still work?

Reply:
Yes, because names are created when the servlet is initialised, it will reply with the names to any URL it can be accessed through.